app/models/piece.rb

```diff
@@ -19,6 +19,20 @@ def move_to!(new_row, new_col)
     update(row_position: new_row, col_position: new_col, moved: true)
   end
 
+  def moving_into_check?(row_dest, col_dest)
+    row_pos = row_position
+    col_pos = col_position
+    # temporarily moving the piece to the new location
+    update(row_position: row_dest, col_position: col_dest)
+    in_check = game.determine_check
+    if in_check
+      update(row_position: row_pos, col_position: col_pos)
+      return true
+    else
+      return false
```

Comment:
How will we call `moving_into_check?`? 
If we are calling it in `valid_move?`, then we should probably move the piece back to the original location even if it is not in check